bloom-housing/bloom
# fix: add backend url validation for url fields

# Pull Request Template

## Issue Overview

This PR addresses #3510


## Description

Adds backend validation for url fields. It should make a difference only for digital application url, as rest has it's own frontend validation, so technically we can't push invalid url field.
Also for it to work i had to fix an issue with application types section reseting on submit (when error). So on edit it took value from listing, even though form value was different.
Also fixed an issue where we could have no digital application selected, but field for url was visible.

## How Can This Be Tested/Reviewed?

On partners site go to edit listing. In application process select `yes` in `is there digital application`, in url field provide invalid url. After submit it should highlight error and display error message. 
I suspect that mocked listings might have mutually exclusive values, so like digitalApplication might have application of "internal" type, but field for `commonDigitalApplication: false` which should result in `externalLink` type (cause thats only case it's not working for me, if i change it and save then it works fine).
Also now after selecting `is there a digital application` -> yes, then `Are you using the common digital application?` -> no, then `is there a digital application` -> no, it should not display url field (i believe that change shouldn't broke apps without common digital application, but worth to double check)

Check other fields if they work fine (to check if backend works i commented out "type="url" prop from fields):
- Open house event
- Building selection criteria
- Settings - preference url
- Settings - preference option url

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
